File: 0745-find-smallest-letter-greater-than-target/0745-find-smallest-letter-greater-than-target.js
/**
 * @param {character[]} letters
 * @param {character} target
 * @return {character}
 */
var nextGreatestLetter = function(letters, target) {

    // if(!letters.includes(target)){
    //     return letters[0];
    // }
    // else{
    //     let index = letters.indexOf(target);
    //     return letters[index + 1];
    // }

    for(let letter of letters){
        if(letter > target){
            return letter
        }
    }
    return letters[0];
};

File: 0745-find-smallest-letter-greater-than-target/0745-find-smallest-letter-greater-than-target.cs
public class Solution {
    public char NextGreatestLetter(char[] letters, char target) {

        foreach (char ch in letters) {
            if (ch > target) {
                return ch;
            }
        }

        return letters[0];

        // int left = 0;
        // int right = letters.Length - 1;

        // while (left <= right) {
        //     int mid = left + (right - left) / 2;

        //     if (letters[mid] <= target) {
        //         left = mid + 1;
        //     } else {
        //         right = mid - 1;
        //     }
        // }

        // If left goes out of bounds, wrap around to the first character
        // return letters[left % letters.Length];
    }
}